aws/resource_aws_batch_compute_environment_test.go

```diff
@@ -0,0 +1,309 @@
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).batchconn
+		name := rs.Primary.Attributes["name"]
+		ce, err := getComputeEnvironment(conn, name)
+		if err != nil {
+			return err
+		}
+		if ce == nil {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		*computeEnv = *ce
+
+		return nil
+	}
+}
+
+func testAccBatchComputeEnvironmentPreCheck(t *testing.T) func() {
+	return func() {
+		testAccPreCheck(t)
+		if os.Getenv("AWS_ACCOUNT_ID") == "" {
+			t.Fatal("AWS_ACCOUNT_ID must be set")
+		}
+	}
+}
+
+func testAccCheckBatchComputeEnvironmentAttributes(computeEnv *batch.ComputeEnvironmentDetail, computeResource *batch.ComputeResource) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if !strings.HasPrefix(*computeEnv.ComputeEnvironmentName, "tf_acctest_batch_compute_environment") {
+			return fmt.Errorf("Bad Compute Environment name: %s", *computeEnv.ComputeEnvironmentName)
+		}
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_batch_compute_environment" {
+				continue
+			}
+			if *computeEnv.ComputeEnvironmentArn != rs.Primary.Attributes["arn"] {
+				return fmt.Errorf("Bad Compute Environment ARN\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["arn"], *computeEnv.ComputeEnvironmentArn)
+			}
+			if *computeEnv.State != rs.Primary.Attributes["state"] {
+				return fmt.Errorf("Bad Compute Environment State\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["state"], *computeEnv.State)
+			}
+
+			if computeResource != nil {
+				if *computeEnv.ComputeResources.MinvCpus != *computeResource.MinvCpus {
+					return fmt.Errorf("Bad Compute Environment MinvCpus\n\t expected: %s\n\tgot: %s\n", *computeEnv.ComputeResources.MinvCpus, *computeResource.MinvCpus)
+				}
+				if *computeEnv.ComputeResources.MaxvCpus != *computeResource.MaxvCpus {
+					return fmt.Errorf("Bad Compute Environment MaxvCpus\n\t expected: %s\n\tgot: %s\n", *computeEnv.ComputeResources.MaxvCpus, *computeResource.MaxvCpus)
+				}
+				if *computeEnv.ComputeResources.DesiredvCpus != *computeResource.DesiredvCpus {
+					return fmt.Errorf("Bad Compute Environment DesiredvCpus\n\t expected: %s\n\tgot: %s\n", *computeEnv.ComputeResources.DesiredvCpus, *computeResource.DesiredvCpus)
+				}
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckBatchComputeEnvironmentDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_batch_compute_environment" {
+			continue
+		}
+		conn := testAccProvider.Meta().(*AWSClient).batchconn
+		describeOpts := &batch.DescribeComputeEnvironmentsInput{
+			ComputeEnvironments: []*string{aws.String(rs.Primary.Attributes["name"])},
+		}
+		resp, err := conn.DescribeComputeEnvironments(describeOpts)
+		if err == nil {
+			if len(resp.ComputeEnvironments) != 0 {
+				return fmt.Errorf("Error: Compute Environment still exists")
+			}
+		}
+		return nil
+	}
+	return nil
+}
+
+const testAccBatchComputeEnvironmentBaseConfig = `
```

Comment:
I'm never sure if such test config/data should be embedded in the test code or kept in a separate file.